### PR TITLE
Treat warnings as errors in tests

### LINKED ISSUE #6638: How should we handle warnings raised during tests?
Currently limited set of warning classes are treated as errror.
https://github.com/chainer/chainer/blob/2615ab68923d0cc55dc3b5c0708f9fdcd48d6b03/setup.cfg#L10-L13

However other warnings are just ignored. Maybe it is better to treat all warnings as error.

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #6638

### CODE REVIEW

**Comment by kmaehashi** [comment]
Let's check how many warnings are raised during test.

Jenkins, test this please.

**Comment by niboshi** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
@kmaehashi Travis has succeeded. Please take a look.

**Comment by niboshi** [comment]
@kmaehashi ping

**Comment by kmaehashi** [comment]
LGTM! Test failures are not related to this PR.

File: setup.cfg [hunk 1]
```diff
@@ -7,10 +7,20 @@ ignore = E741,W503,W504,E241,E226
...
+                ignore:numpy\.ufunc size changed, may indicate binary incompatibility:RuntimeWarning
+                ignore:numpy\.dtype size changed, may indicate binary incompatibility:RuntimeWarning
+                ignore:ChainerX core binary is built in debug mode:UserWarning
+                ignore:Accelerate has been detected as a NumPy backend library:UserWarning:chainer[.*]
+                ignore::chainer.warnings.PerformanceWarning
                 ignore::FutureWarning:chainer\.utils\.experimental
-                error::DeprecationWarning
-                error::PendingDeprecationWarning
+                # TODO(niboshi): Fix the warnings below
+                ignore:Some of grads are ignored by chainer\.backward:RuntimeWarning
```
Comment (toslunar) [inline setup.cfg]: It seems fixed by #7762.

Comment (niboshi) [reply setup.cfg]: Thanks, fixed.

**Comment by niboshi** [comment]
PTAL

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by niboshi** [comment]
Hmm, why is cupy 7.0.0b1 used?
```
2019-07-25 12:28:57.964569 STDOUT 2301] | INTERNALERROR> UserWarning:
-- | --
2019-07-25 12:28:57.964573 STDOUT 2301] | INTERNALERROR> --------------------------------------------------------------------------------
2019-07-25 12:28:57.964574 STDOUT 2301] | INTERNALERROR> CuPy (cupy) version 7.0.0b1 may not be compatible with this version of Chainer.
2019-07-25 12:28:57.964575 STDOUT 2301] | INTERNALERROR> Please consider installing the supported version by running:
2019-07-25 12:28:57.964576 STDOUT 2301] | INTERNALERROR>   $ pip install 'cupy==7.0.0b2'
2019-07-25 12:28:57.964577 STDOUT 2301] | INTERNALERROR>
2019-07-25 12:28:57.964599 STDOUT 2301] | INTERNALERROR> See the following page for more details:
2019-07-25 12:28:57.964600 STDOUT 2301] | INTERNALERROR>   https://docs-cupy.chainer.org/en/latest/install.html
2019-07-25 12:28:57.964601 STDOUT 2301] | INTERNALERROR> --------------------------------------------------------------------------------
```

**Comment by kmaehashi** [comment]
I've re-triggered the docker image build.
https://ci.preferred.jp/chainer-ci-prep.py27and35/20846/
https://ci.preferred.jp/chainer-ci-prep.py37/20847/

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
@kmaehashi ping

**Comment by niboshi** [comment]
@kmaehashi 
What's the status of this PR?
I want to merge it as soon as possible.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
There are still warnings I'm afraid that we need to address. The log is huge so it's difficult to go through them all from the web browser but some are obvious such as the `Some of grads are ignored` raised by `backward`. If there are too many different kinds, how about adding them to the whitelist and then tackle them one by one in separate PRs?

Noted that pfnCI is failing on CuPy version but the above is about the current Jenkins results.

**Comment by niboshi** [comment]
Added an ignore entry.
I hope all of the warnings will vanish.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Another `UserWarning` was still lurking on pfnCI for a maybe-incompatible CuPy version 😞 Could you add it to the list?

**Comment by hvy** [comment]
I know noticed that some jobs on pfnCI are succeeding although the above `UserWarning` is raised?

**Comment by niboshi** [comment]
@hvy Which one?

**Comment by hvy** [comment]
I didn't check all jobs but most of them are raising some sort of warning, but passing. Only `pfn-public-ci/chainer.py37.{cpu,gpu}` are failing as expected.

**Comment by hvy** [comment]
CuPy builds have been updated and the CI is now expected to pass... Jenkins, test this please.

**Pull request merged**